src/ProjectTemplates/Web.ProjectTemplates/content/StarterWeb-CSharp/Program.cs

```diff
@@ -2,25 +2,139 @@
+using Microsoft.AspNetCore.Authorization;
+#endif
+using Microsoft.AspNetCore.Builder;
 using Microsoft.AspNetCore.Hosting;
+#if (IndividualLocalAuth)
+using Microsoft.AspNetCore.Identity;
+#endif
+#if (OrganizationalAuth)
+using Microsoft.AspNetCore.Mvc.Authorization;
+#endif
+#if (IndividualLocalAuth)
+using Microsoft.EntityFrameworkCore;
+#endif
 using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting;
 using Microsoft.Extensions.Logging;
+#if (MultiOrgAuth)
+using Microsoft.IdentityModel.Tokens;
+#endif
+#if (GenerateGraph)
+using Graph = Microsoft.Graph;
+#endif
+#if (IndividualLocalAuth)
+using Company.WebApplication1.Data;
+#endif
 
-namespace Company.WebApplication1
+var builder = WebApplication.CreateBuilder(args);
+
+// Add services to the container.
+#if (IndividualLocalAuth)
+builder.Services.AddDbContext<ApplicationDbContext>(options =>
+#if (UseLocalDB)
+    options.UseSqlServer(
+        builder.Configuration.GetConnectionString("DefaultConnection")));
+#else
+    options.UseSqlite(
+        builder.Configuration.GetConnectionString("DefaultConnection")));
+#endif
+builder.Services.AddDatabaseDeveloperPageExceptionFilter();
+
+builder.Services.AddDefaultIdentity<IdentityUser>(options => options.SignIn.RequireConfirmedAccount = true)
+    .AddEntityFrameworkStores<ApplicationDbContext>();
+#elif (OrganizationalAuth)
+#if (GenerateApiOrGraph)
+var initialScopes = builder.Configuration.GetValue<string>("DownstreamApi:Scopes")?.Split(' ');
+
+#endif
+builder.Services.AddAuthentication(OpenIdConnectDefaults.AuthenticationScheme)
+#if (GenerateApiOrGraph)
+    .AddMicrosoftIdentityWebApp(builder.Configuration.GetSection("AzureAd"))
+        .EnableTokenAcquisitionToCallDownstreamApi(initialScopes)
+#if (GenerateApi)
+            .AddDownstreamWebApi("DownstreamApi", builder.Configuration.GetSection("DownstreamApi"))
+#endif
+#if (GenerateGraph)
+            .AddMicrosoftGraph(builder.Configuration.GetSection("DownstreamApi"))
+#endif
+            .AddInMemoryTokenCaches();
+#else
+    .AddMicrosoftIdentityWebApp(builder.Configuration.GetSection("AzureAd"));
+#endif
+#elif (IndividualB2CAuth)
+#if (GenerateApi)
+var initialScopes = builder.Configuration.GetValue<string>("DownstreamApi:Scopes")?.Split(' ');
+
+#endif
+builder.Services.AddAuthentication(OpenIdConnectDefaults.AuthenticationScheme)
+#if (GenerateApi)
+    .AddMicrosoftIdentityWebApp(builder.Configuration.GetSection("AzureAdB2C"))
+        .EnableTokenAcquisitionToCallDownstreamApi(initialScopes)
+            .AddDownstreamWebApi("DownstreamApi", builder.Configuration.GetSection("DownstreamApi"))
+            .AddInMemoryTokenCaches();
+#else
+    .AddMicrosoftIdentityWebApp(builder.Configuration.GetSection("AzureAdB2C"));
+#endif
+#endif
+#if (OrganizationalAuth)
+
+builder.Services.AddControllersWithViews(options =>
+{
+    var policy = new AuthorizationPolicyBuilder()
```

Comment:
@pranavkm @HaoK I thought we did this with middleware now. Why does this template still use the authorize filter?

Reply:
Yeah that looks strange. Seems like a relic?